applications/utils.py

```diff
@@ -25,7 +25,4 @@ def check_eligibility_to_skip_steps(bootcamp_application):
     Args:
         bootcamp_application (BootcampApplication)
     """
-    return (
-        bootcamp_application.user.profile.can_skip_application_steps
-        and bootcamp_application.bootcamp_run.allows_skipped_steps
-    )
+    return bootcamp_application.bootcamp_run.allows_skipped_steps
```

Comment:
Currently, if I register as a new user without  `user.profile.can_skip_application_steps=True`, I cannot apply to a bootcamp where `allows_skipped_steps=True` because it is filtered out from the dropdown list.

It seems `allows_skipped_steps=True` is meant to be used together with `user.profile.can_skip_application_steps=True` to allow alumni to skip directly to payment for bootcamps that do have submission steps.

I think this issue is supposed to make it possible for any users applying to stepless bootcamps (no submission steps defined for the bootcamp) to go straight to payment even if `allows_skipped_steps=False` (because there aren't any submission steps).

You can probably start implementing it this way by doing something like this:

`applications.api.derive_application_state`:
```python
def derive_application_state(
    bootcamp_application,
):  # pylint: disable=too-many-return-statements
    if not is_user_info_complete(bootcamp_application.user):
        return AppStates.AWAITING_PROFILE_COMPLETION.value
    # New code starts here    
    if bootcamp_application.bootcamp_run.application_steps.count() === 0:
        return AppStates.AWAITING_PAYMENT.value
```

and with changes to the `ApplicationDashboardPage` in comments below